command_line_assistant/daemon/http/query.py

```diff
@@ -83,6 +83,14 @@ def submit(payload: dict, config: Config) -> str:
         raise RequestFailedError(
             f"Communication error with the server: {str(exc)}. Please try again in a few minutes."
         ) from exc
+    except OSError as exc:
+        # If the exception string contains the standard RHSM cert path, assume
+        # the system is not registered and raise a very specific error message.
+        if "/etc/pki/consumer/cert.pem" in str(exc):
```

Comment:
Using the `str` is probably fine, but it may be better to look at the `strerror` attribute directly if it's available. The `OSError` constructor sets different attributes based on what was passed in.
```suggestion
        if "/etc/pki/consumer/cert.pem" in exc.strerror.casefold():
```

Reply:
`strerror` is optional according to the type annotations. Changing the check to compare `strerror` actually causes the corresponding test to fail because initializing the mock `OSError` with a string value does not set `strerror`. This leads me to question whether relying on `strerror` is too brittle. If it's not always set, we have to have fallback code to check the string representation anyway, so we might as well check the string representation to begin with. In a real world scenario, will the `OSError` always be constructed with a `strerror` value?

Reply:
Yeah, we can hardcode them in the code, or remove it from the config file and make it "hidden"

Reply:
I think we can explore removing the config file values entirely, but not in the scope of this PR.